### PR TITLE
Bring together mandotory attributes validation and visited groups

### LINKED ISSUE #7660: Bring together mandotory attributes validation and visited groups
At the moment the mandatory attributes validation takes place after putting a product to the cart. It should be also done after the group has been visited and left.

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
-    return false;
+  showRequiredMessage(): Observable<boolean> {
+    return this.configUtils
+      .isCartEntryOrGroupVisited(this.owner, this.groupId)
+      .pipe(
+        map((result) => {
+          if (
+            result &&
+            this.attribute.required &&
+            this.attribute.incomplete &&
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts]: This line is not needed, right? If an attribute is required and userInput is blank, we need to show the message. (Or am I wrong?)

Comment (ChristophHi) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts]: And if we merge this one _after_ the PR dealing with numeric attributes, we need to cover UiType.NUMERIC as well

File: projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    expect(classUnderTest).toBeTruthy();
+  });
+
+  it('should return false because the product has not been added to the cart and the current group was not visited', () => {
+    let result: boolean;
+    const subscription = classUnderTest
+      .isCartEntryOrGroupVisited(owner, groupId)
+      .subscribe((data) => (result = Boolean(data)));
+    expect(result).toBe(false);
+    subscription.unsubscribe();
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.spec.ts]: I guess shorter and more readable to directly call .unsubscribe after subscribe, then const subscription is not needed

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.spec.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.spec.ts]: good point! I went even further and extracted a getActualResult method where the whole magix takes place. :)

File: projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts]: Good hint! I completely forgot about the numeric input validation.

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**